src/main/java/org/folio/orders/utils/HelperUtils.java

```diff
@@ -70,10 +76,25 @@ public static JsonObject verifyAndExtractBody(Response response) {
     return response.getBody();
   }
 
+  /**
+   * If response http code is {@code 404}, empty {@link JsonObject} is returned. Otherwise verifies if response is successful and extracts body
+   * In case there was failed attempt to delete PO or particular PO line, the sub-objects might be already partially deleted.
+   * This check allows user to retry DELETE operation
+   *
+   * @param response response to verify
+   * @return empty {@link JsonObject} if response http code is {@code 404}, otherwise verifies if response is successful and extracts body
+   */
+  private static JsonObject verifyAndExtractBodyIfFound(Response response) {
+    if (response.getCode() == 404) {
```

Comment:
Yes, you are right. The DELETE request to storage always return 204. However in case most of the data of the PO line already deleted but something failed, the PO line won't be deleted from storage. And it would still keep references to deleted sub-objects. In such case I believe some popup with error will be shown to the end user on UI. There are 2 cases possible:
1. User just tries to click delete again, the logic will just retrieve PO line again and all DELETE requests would produce 204. We are ok.
2. User decides to refresh the screen and UI app will send GET PO line request to mod-orders. This will result to attempt to GET all the sub-objects and for those already deleted `404` would be returned. In this case logic will not return composite PO line back to client and user won't be able to retry deletion.

So the check above is basically for the second case. Probably the java doc I added is not so good :)